### PR TITLE
Cleans up some vintage shitcode

### PR DESCRIPTION
**Pull request by Cyberboss**

### CODE REVIEW

File: code/game/machinery/lightswitch.dm [hunk 1]
```diff
@@ -11,21 +11,18 @@
...
-		if(otherarea)
-			src.area = locate(text2path("/area/[otherarea]"))
-
-		if(!name)
-			name = "light switch ([area.name])"
-
-		src.on = src.area.lightswitch
-		updateicon()
+	if(otherarea)
+		area = locate(text2path("/area/[otherarea]"))
```
Comment (RemieRichards) [inline code/game/machinery/lightswitch.dm]: why is otherarea not just the entire typepath?

Comment (Cyberboss) [inline code/game/machinery/lightswitch.dm]: ¯\_(ツ)_/¯

Comment (RemieRichards) [inline code/game/machinery/lightswitch.dm]: this is your cue to fix it.

Comment (Cyberboss) [inline code/game/machinery/lightswitch.dm]: reee im in the middle of a big thing

**Pull request merged**